examples/terraform/azure/output.tf

```diff
@@ -91,6 +91,8 @@ output "kubeone_workers" {
           # assignAvailabilitySet = true/false
           securityGroupName = azurerm_network_security_group.sg.name
           assignPublicIP    = true
+          imageReference    = var.os != "rhel" ? var.image_references[var.os].image : null
+          imagePlan         = length(var.image_references[var.os].plan) > 0 && var.os != "rhel" ? var.image_references[var.os].plan[0] : null
```

Comment:
I'd suggest to change this in variables.tf instead, like this:

```diff
diff --git examples/terraform/azure/variables.tf examples/terraform/azure/variables.tf
index 2e12c9ef..508ba4dd 100644
--- examples/terraform/azure/variables.tf
+++ examples/terraform/azure/variables.tf
@@ -159,17 +159,8 @@ variable "image_references" {
     }
 
     rhel = {
-      image = {
-        publisher = "RedHat"
-        offer     = "rhel-byos"
-        sku       = "rhel-lvm85"
-        version   = "8.5.20220316"
-      }
-      plan = [{
-        name      = "rhel-lvm85"
-        publisher = "redhat"
-        product   = "rhel-byos"
-      }]
+      image = null
+      plan = null
       ssh_username = "rhel-user"
       worker_os    = "rhel"
     }
```

Reply:
This is needed for control plane, so we can't do this.